src/Resolvers/DomainTenantResolver.php

```diff
@@ -28,18 +28,30 @@ class DomainTenantResolver extends Contracts\CachedTenantResolver
 
     public function resolveWithoutCache(...$args): Tenant
     {
-        /** @var Domain $domain */
-        $domain = config('tenancy.domain_model')::where('domain', $args[0])->first();
+        $domain = $args[0];
 
-        if ($domain) {
-            static::$currentDomain = $domain;
+        /** @var Tenant|null $tenant */
+        $tenant = config('tenancy.tenant_model')::query()
+            ->whereHas('domains', function ($query) use ($domain) {
+                $query->select(['tenant_id', 'domain'])->where('domain', $domain);
+            })
+            ->with('domains')
+            ->first();
 
-            return $domain->tenant;
+        if ($tenant) {
+            $this->tenantIdentified($tenant, ...$args);
```

Comment:
I think that rather than calling `tenantIdentified()` from this method, the logic that sets the current domain should be in a separate method, one that's called both from here and from `tenantIdentified()`.

Seems weird to never call "tenant identified" when cache is not used in the other resolvers, but use it in this one. The methods should have more specific roles.